code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/api/action/AddServiceLinkActionHandler.java

```diff
@@ -16,14 +15,12 @@
 
 @Named
 public class AddServiceLinkActionHandler implements ActionHandler {
-    @Inject
-    ServiceConsumeMapDao consumeMapDao;
 
     @Inject
-    ObjectManager objectManager;
+    JsonMapper jsonMapper;
 
     @Inject
-    ObjectProcessManager objectProcessManager;
+    ServiceDiscoveryService sdService;
```

Comment:
Along the same lines as the comment about LoadBalancerServiceImpl.  It scares me to see API level code invoking backend classes.  I like to see a clear separation.